gensim/models/sent2vec.py

```diff
@@ -419,6 +419,8 @@ def __init__(self, sentences=None, vector_size=100, lr=0.2, lr_update_rate=100,
         self.minn = minn
         self.maxn = maxn
         self.dropoutk = dropoutk
+        self.hidden = np.zeros(vector_size, dtype=np.float32)
```

Comment:
This has to match `REAL_t` (if they ever diverge in the future, you'll get segfaults).

Also be careful around multithreading. Each thread will need its own "scratch space", so they don't overwrite the same array at the same time.

Reply:
I understand that the threads will need different hidden and grad arrays, but what about updating the weights? Won't all threads have access to the same matrices of neural weights? So it's possible that they may be overwriting the same weights at the same time. Is there a workaround for this?

Reply:
Great point! You are correct, but that's called a feature :)

It's a technique called HogWild and it's used in algorithms too. Word2vec and doc2vec also overwrite the same parameter matrices.

The idea is that the hardware provides enough atomicity (updating a single float / a SIMD operation / cache line) to make it not crash / go horribly wrong. And then the threads stepping on each other's toes cancel out, it turns out it's not a big deal (and surprisingly, sometimes even helps!).